gfx/common/win32_common.c

```diff
@@ -1172,6 +1175,33 @@ static LRESULT CALLBACK wnd_proc_common_dinput_internal(HWND hwnd,
 
    switch (message)
    {
+      case WM_IME_ENDCOMPOSITION:
+            input_keyboard_event(true, 1,  0x80000000, 0, RETRO_DEVICE_KEYBOARD); 
+            break;
+      case WM_IME_COMPOSITION:
+      {  
+         HIMC    hIMC = ImmGetContext(hwnd);
+         unsigned gcs = lparam & (GCS_COMPSTR|GCS_RESULTSTR);	
+         if( gcs )
+         {
+            int i;
+            wchar_t wstr[4]={0,};
+            int len1 = ImmGetCompositionStringW(hIMC, gcs, wstr, 4);		
+            wstr[2] = wstr[1];
+            wstr[1] = 0;
+            if (len1<=0 || len1>4 )  break;
+            for( i=0; i<len1; i=i+2)					        
+            {
+               char  utf8[8]={0,};
+               int len2 = WideCharToMultiByte( CP_UTF8, 0, wstr+i, -1, utf8, 8, NULL, NULL );	
```

Comment:
Can you maybe use the UTF to char conversion functions in libretro-common instead? You can find equivalent functions to WideCharToMultiByte in libretro-common/encodings/encoding_utf.c I believe.